core/primitives/tipset/tipset_key.cpp

```diff
@@ -92,15 +96,23 @@ namespace fc::primitives::tipset {
     return common::hex_lower(hash_);
   }
 
-  std::string TipsetKey::toPrettyString() const {
-    return fmt::format("{{{}}}", fmt::join(cids_, ","));
+  std::string TipsetKey::cidsStr(std::string_view sep) const {
+    std::string s;
+    auto first{true};
+    for (auto &cid : cids_) {
+      s += cid.toHex();
+      if (first) {
+        first = false;
+      } else {
```

Comment:
What about `for (int i = 1; i < cids_.size(); ++i)` ?

Reply:
I fear duplicating code, so i wrote that loop